<?xml version="1.0"?>
<launch>
  <!-- Arguments -->
  <arg name="move_forward_only"   default="false"/>
  <arg name="waypoint_navigation" default="true"/>

  <arg name="map_name"          default="tsudanuma_2_19" doc="map_name type [tsudanuma_2_19, tsudanuma, tsukuba]"/>
  <arg name="mcl_map_file"        default="$(find raspicat_waypoint_navigation)/config/maps/$(arg map_name)/map_$(arg map_name).yaml"/>
  <arg name="move_base_map_file"  default="$(find raspicat_waypoint_navigation)/config/maps/for_move_base/map_$(arg map_name).yaml"/>
  <arg name="waypoint_yaml_file"  default="$(find raspicat_waypoint_navigation)/config/waypoint/waypoint.yaml"/>
  <arg name="mcl"                 default="amcl" doc="model type [amcl, emcl2, mcl_ros]"/>

  <arg name="x_mcl"       default="-0.309"/>
  <arg name="y_mcl"       default="0.112"/>
  <arg name="a_mcl"       default="-0.013"/>

  <!-- Amcl -->
  <group if="$(eval mcl=='amcl')">
    <include file="$(find raspicat_waypoint_navigation)/launch/amcl.launch">
      <arg name="initial_pose_x"  value="$(arg x_mcl)"/>
      <arg name="initial_pose_y"  value="$(arg y_mcl)"/>
      <arg name="initial_pose_a"  value="$(arg a_mcl)"/>
    </include>
  </group>

  <!-- Map_server -->
  <node pkg="map_server" name="mcl_map_server" type="map_server" args="$(arg mcl_map_file)"/>

  <node pkg="map_server" name="move_base_map_server" type="map_server" args="$(arg move_base_map_file)">
    <remap from="map" to="/move_base_map"/>
    <remap from="static_map" to="/move_base_static_map"/>
  </node>

  <!-- Move_base -->
  <include file="$(find raspicat_waypoint_navigation)/launch/move_base.launch">
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    <arg name="waypoint_nav" default="$(arg waypoint_navigation)"/>
  </include>

  <!-- Raspicat speak -->
  <!-- <node pkg="raspicat_speak" type="raspicat_speak_node" name="raspicat_speak_node" 
        clear_params="true" respawn="true" output="screen">
    <rosparam command="load" file="$(find raspicat_waypoint_navigation)/config/raspicat_speak/speak_list.yaml"/>
    <rosparam command="load" file="$(find raspicat_waypoint_navigation)/config/raspicat_speak/voice_config.yaml"/>
  </node> -->

  <!-- Waypoint_nav -->
  <node if="$(arg waypoint_navigation)" pkg="raspicat_waypoint_navigation" name="WaypointNav_node" type="WaypointNav_node" output="screen">
    <rosparam command="load" file="$(arg waypoint_yaml_file)"/>
    <param name="initial_pose_x"  value="$(arg x_mcl)"/>
    <param name="initial_pose_y"  value="$(arg y_mcl)"/>
    <param name="initial_pose_a"  value="$(arg a_mcl)"/>
  </node>
	
	<!-- Obstacle detector -->
	<rosparam command="load" file="$(find raspicat_waypoint_navigation)/config/param/obstacle_detector.yaml"/>
  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node" clear_params="true" output="screen"/>
</launch>